### PR TITLE
Bug 1827414: e2e: Add ibmcloud provider, skips and fixes

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

**Comment by csrwng** [comment]
@derekwaynecarr @smarterclayton this should be ready

File: test/extended/prometheus/prometheus_builds.go [hunk 1]
```diff
@@ -48,6 +49,11 @@ var _ = g.Describe("[sig-instrumentation][sig-builds][Feature:Builds] Prometheus
 
 	g.Describe("when installed on the cluster", func() {
 		g.It("should start and expose a secured proxy and verify build metrics", func() {
+			if e2e.TestContext.Provider == ibmcloud.ProviderName {
+				e2eskipper.Skipf("Prometheus in IBM ROKS clusters does not collect metrics from the OpenShift controller manager because it lives outside of the cluster. " +
+					"The openshift_build_total metric expected by this test is reported by the OpenShift controller manager")
```
Comment (derekwaynecarr) [inline test/extended/prometheus/prometheus_builds.go:54]: please link to to a JIRA tracking if this should move to openshift state metrics.

File: test/extended/util/annotate/rules.go [hunk 2]
```diff
@@ -197,7 +197,56 @@ var (
...
+			// Currently ibm-master-proxy-static and imbcloud-block-storage-plugin tolerate all taints
+			// https://bugzilla.redhat.com/show_bug.cgi?id=1825027
+			`\[Feature:Platform\] Managed cluster should ensure control plane operators do not make themselves unevictable`,
+
+			// Prometheus is not reporting router metrics
+			// https://bugzilla.redhat.com/show_bug.cgi?id=1825029
+			`\[Feature:Prometheus\]\[Conformance\] Prometheus when installed on the cluster should provide ingress metrics`,
+			`\[Conformance\]\[Area:Networking\]\[Feature:Router\] The HAProxy router should enable openshift-monitoring to pull metrics`,
+
+			// ROKS cluster role bindings don't match expected results
```
Comment (derekwaynecarr) [inline test/extended/util/annotate/rules.go:242]: note to self:
no action needed, bug states that ROKS includes additional bindings.

File: test/extended/util/annotate/rules.go [hunk 3]
```diff
@@ -197,7 +197,56 @@ var (
...
+			// https://bugzilla.redhat.com/show_bug.cgi?id=1825030
+			`TestAuthorizationResourceAccessReview should succeed`,
+
+			// Test does not allow enough time for the pods to be created before deleting them
```
Comment (derekwaynecarr) [inline test/extended/util/annotate/rules.go:246]: note:
this is a poor upstream test that is just time-rather than state based.

**Comment by csrwng** [comment]
Thx @derekwaynecarr, comments addressed.

**Comment by csrwng** [comment]
/retitle Bug 1827414: e2e: Add ibmcloud provider, skips and fixes

**Pull request merged**